lib/helpers.bash

```diff
@@ -14,16 +14,18 @@ function _command_exists ()
   type "$1" &> /dev/null ;
 }
 
-# Helper function loading various enable-able files
-function _load_bash_it_files() {
+# Helper function listing various enable-able files to be sourced
+# The files need to be sourced in global scope to preserve scope of 'declare'
+function _list_bash_it_files() {
   subdirectory="$1"
   if [ -d "${BASH_IT}/${subdirectory}/enabled" ]
   then
     FILES="${BASH_IT}/${subdirectory}/enabled/*.bash"
-    for config_file in $FILES
+
+    for _bash_it_config_file in $FILES
     do
-      if [ -e "${config_file}" ]; then
-        source $config_file
+      if [ -e "${_bash_it_config_file}" ]; then
+        _bash_it_list_bash_it_files_return+=("$_bash_it_config_file")
```

Comment:
I like the idea of where you're headed with this. What about making this a local variable though, and then returning it via something like `echo "${accumulator[*]}"`. I made this change locally and modified `bash_it.sh` to look like this:

```bash
for file_type in "aliases" "plugins" "completion"
do
  for _bash_it_enabled_config_file in $(_list_bash_it_files "$file_type"); do
    . "$_bash_it_enabled_config_file"
  done
done
```

I've played a little bit with `_make_reload_alias` below, but I can't get it to render the above block in a way that executes as expected. I haven't done multi-line aliases like that, though, so I'm sure I'm missing something obvious.

Reply:
That's better indeed.

I used a global array because printing a list of file names is impossible to do correctly without some escaping (*nix file names can contain tabs, newlines and spaces), but in that case, since we're dealing with filenames we control and using relative paths, we don't need to worry about that.

Reply:
Actually, my bad, we're dealing with absolute paths.

The enabled files are listed with

`FILES="${BASH_IT}/${subdirectory}/enabled/*.bash"`

So the proposed solution will break if bash-it is installed under a directory  that contains a char that's also in IFS, meaning there's no way to make it behave 100% correctly.

A possible, clean-enough solution is to make the list of returned files relative to $BASH_IT. I've done so in my newer commit.